### PR TITLE
Add a transformer to add 'Operand Label' to resources

### PR DESCRIPTION
**Pull request by nikhil-thomas**
- [x] Add a transformer which adds `operator.tekton.dev/operand-name` label
    eg: `operator.tekton.dev/operand-name: tektoncd-pipelines` to all resources in tektoncd pipeline payload
- [ ] add this new transformer to all tektoncd/operator components (operands)
- [ ] modify e2e tests to use this label instead of labels added and managed by component projects itself

Signed-off-by: Nikhil Thomas <nikthoma@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
- adds `operator.tekton.dev/operand-name: <operand-name>` label to all resources created by operator (eg: `[operator.tekton.dev/operand-name](http://operator.tekton.dev/operand-name): tektoncd-pipelines`)
```

### CODE REVIEW

File: pkg/reconciler/common/transformer_injectlabel.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+func InjectLabelOverwriteExisting(newLabels labels.Set, skipChecks ...mf.Predicate) mf.Transformer {
+	preserverExisting := false
+	return injectLabel(newLabels, preserverExisting, skipChecks...)
+}
+
+func injectLabel(newLabels labels.Set, preserverExisting bool, skipChecks ...mf.Predicate) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		for _, skipCheck := range skipChecks {
+			//fmt.Println(u.GetKind(), u.GetName(), skipCheck(u))
```
Comment (pradeepitm12) [inline pkg/reconciler/common/transformer_injectlabel.go:44]: Nit: we can remove this.

File: pkg/reconciler/common/transformer_injectlabel.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+func replaceAllowed(preserveExisting bool, existingLabels map[string]string, key string) bool {
+	if !preserveExisting {
+		return true
+	}
+	_, ok := existingLabels[key]
+	if !ok {
+		return true
+	}
+	// preserverExisting == true && key exists in current labels
```
Comment (pradeepitm12) [inline pkg/reconciler/common/transformer_injectlabel.go:72]: nit: and this

File: pkg/reconciler/common/transformer_injectlabel.go (hunk 1 above)
Comment (nikhil-thomas) [reply pkg/reconciler/common/transformer_injectlabel.go]: thanks 🧑‍💻 

fixed 👍

File: pkg/reconciler/common/transformer_injectlabel.go (hunk 2 above)
Comment (nikhil-thomas) [reply pkg/reconciler/common/transformer_injectlabel.go]: this one is a comment explaing the last `return false` 🧑‍💻 

i think i need to add a `then return false` to the comment.

done 👍

**Review by vdemeester — APPROVED** [review]

**Review by vdemeester — APPROVED** [review]
/lgtm

**Pull request merged**